src/kbmod/result_set.py

```diff
@@ -0,0 +1,502 @@
+import copy
+import csv
+import math
+import numpy as np
+import kbmod.search as kb
+
+class SharedTools:
+    """
+    This class manages tools that are shared by the classes Interface and
+    PostProcess.
+    
+    Legacy approach. Soon to be deprecated.
+    """
+
+    def __init__(self):
+        return
+
+    def gen_results_dict(self):
```

Comment:
do we still need SharedTools.gen_results_dict anymore? I'm fine with keeping it around for posterity's sake if we still need it for some parts of the code :)

Reply:
Very short term: Yes. Most of the code still uses it, but will be cleaned up in a second PR.
Short term: Yes. We need it for backwards compatibility with code that might use part of analysis_utils.py
Long term: I'm hoping we can deprecate and remove it.